src/embed/types.rs

```diff
@@ -46,13 +47,16 @@ pub struct OpenTransactionResponse {
 pub struct CommitTransactionRequest {
     #[serde(rename = "transactionId")]
     pub transaction_id: u32,
+    #[serde(rename = "generateDiffs")]
+    pub generate_diffs: bool,
 }
 
 #[derive(Debug, Deserialize, Serialize)]
 pub struct CommitTransactionResponse {
     // Note: the field is named "ref" in go but "ref" is a reserved word in rust.
     #[serde(rename = "ref")]
     pub hash: String,
+    pub diffs: BTreeMap<String, Vec<String>>,
```

Comment:
Can you ensure this isn't significantly increasing the size of the js generated to serialize? I think it might be the first occurrence of BTreeMap going over this boundary.

Reply:
It is the first, I would also be fine going through a `Vec<(String, String)>`. 

Let me try to figure out how it affects size.

Reply:
```
With Map:
replicache_client_bg.wasm: 773140
replicache_client_bg.wasm.br: 209530

With Entries:
replicache_client_bg.wasm: 774323
replicache_client_bg.wasm.br: 209972

Diff 
replicache_client_bg.wasm: 1183
replicache_client_bg.wasm.br: 442
```

Using a `HashMap<String, Vec<String>>` is more efficient than `Vec<(String, Vec<String>)>`